core/shared/src/main/scala/scalaz/zio/IO.scala

```diff
@@ -858,19 +865,19 @@ object IO extends Serializable {
   final def suspend[E, A](io: => IO[E, A]): IO[E, A] = new Suspend(() => io)
 
   /**
-   * Terminates the fiber executing this action, running all finalizers.
+   * Interrupts the fiber executing this action, running all finalizers.
    */
-  final def terminate: IO[Nothing, Nothing] = terminate0(Nil)
+  final def interrupt: IO[Nothing, Nothing] = new Terminate(None, Nil)
```

Comment:
This should actually be called `terminate`, because a fiber cannot (currently) interrupt itself, but it can terminate itself. There's an implicit law that:

```scala
io.terminate(t) === io.map(throw t)
```

Also now that we are requiring a `Throwable` for terminate (as would be expected by the above law), we should delete the no-parameter version of `terminate`. 

Interruption on a `Fiber` can still have a no-parameter variant, because interruption does not have to have a specified cause.

Reply:
Yeah I initially removed it and then put it back because it was called in RTS in case of self interruption (without throwable). That is also why I added Option[_]. I wasn't really sure what to do with it.
If we have another way to handle it (as suggested in your other comments), this can go.